<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>762</width>
    <height>369</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>30</y>
      <width>331</width>
      <height>311</height>
     </rect>
    </property>
    <property name="title">
     <string>输入</string>
    </property>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>190</y>
       <width>51</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>使用算法</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="inti_3">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>60</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>7</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="target_8">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>140</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>8</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QComboBox" name="algorithm_type">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>210</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="editable">
      <bool>false</bool>
     </property>
     <property name="sizeAdjustPolicy">
      <enum>QComboBox::AdjustToContentsOnFirstShow</enum>
     </property>
     <item>
      <property name="text">
       <string>A*-Euclid</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>A*-HVdist</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>A*-Simple</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>BFS-discouraged</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="start">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>260</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>开始</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="target_7">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>140</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>7</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="restart">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>260</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>重置</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="inti_8">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>140</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>3</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="target_5">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>100</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>5</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="inti_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>100</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="inti_5">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>100</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>5</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="target_9">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>140</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="inti_7">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>140</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>4</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="target_4">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>100</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>4</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="target_1">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>60</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>1</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="target_2">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>60</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>2</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>30</y>
       <width>54</width>
       <height>12</height>
      </rect>
     </property>
     <property name="text">
      <string>结束位置</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="inti_9">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>140</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>2</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="inti_6">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>100</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>1</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="inti_2">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>60</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>6</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="target_6">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>100</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>6</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="target_3">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>60</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>3</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="inti_1">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>8</string>
     </property>
     <property name="maxLength">
      <number>1</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>30</y>
       <width>54</width>
       <height>12</height>
      </rect>
     </property>
     <property name="text">
      <string>起始位置</string>
     </property>
    </widget>
    <widget class="QPushButton" name="creat_data">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>260</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>随机生成</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>30</y>
      <width>331</width>
      <height>151</height>
     </rect>
    </property>
    <property name="title">
     <string>输出</string>
    </property>
    <widget class="QScrollArea" name="scrollArea">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>321</width>
       <height>121</height>
      </rect>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>302</width>
        <height>178</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="time_label">
         <property name="text">
          <string>时间：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="step_label">
         <property name="text">
          <string>解步长：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="expanded_label">
         <property name="text">
          <string>已扩展节点：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="to_expand_label">
         <property name="text">
          <string>待扩展节点：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="waixianlv">
         <property name="text">
          <string>外显率：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="fenzhiyinshu">
         <property name="text">
          <string>有效分支因数：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>过程展示：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="out_show">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>180</y>
      <width>331</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>过程展示</string>
    </property>
    <widget class="QLabel" name="output_9">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>120</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="output_3">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="output_4">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>80</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="output_8">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>120</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="output_6">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>80</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="output_5">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>80</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="output_1">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="output_7">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>120</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="output_2">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>40</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="speed_num">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>40</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>5</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
     <property name="textInteractionFlags">
      <set>Qt::NoTextInteraction</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>10</y>
       <width>51</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>速度控制</string>
     </property>
    </widget>
    <widget class="QSlider" name="speed">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>10</y>
       <width>21</width>
       <height>71</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>120</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>跳转到</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="Go_Where">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>120</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="maxLength">
      <number>3</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="Go_Step">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>120</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>GO</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>speed</sender>
   <signal>valueChanged(int)</signal>
   <receiver>speed_num</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>720</x>
     <y>305</y>
    </hint>
    <hint type="destinationlabel">
     <x>775</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>restart</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>ButtonClickedRestore()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>260</x>
     <y>265</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>start</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>ButtonClickedStart()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>315</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Go_Step</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>BottonClickShow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>695</x>
     <y>309</y>
    </hint>
    <hint type="destinationlabel">
     <x>380</x>
     <y>184</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>creat_data</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>BottonClickRandom()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>305</y>
    </hint>
    <hint type="destinationlabel">
     <x>380</x>
     <y>184</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>ButtonClickedRestore()</slot>
  <slot>ButtonClickedStart()</slot>
  <slot>BottonClickShow()</slot>
  <slot>BottonClickRandom()</slot>
 </slots>
</ui>
